crates/shared/src/bad_token/token_owner_finder/seasolver.rs

```diff
@@ -0,0 +1,45 @@
+use super::TokenOwnerProposing;
+use anyhow::Result;
+use ethcontract::H160;
+
+pub struct SeasolverTokenOwnerFinder {
+    owners: Vec<H160>,
+}
+
+impl SeasolverTokenOwnerFinder {
+    pub fn new(owners: Vec<H160>) -> Self {
+        Self { owners }
+    }
+}
+
+#[async_trait::async_trait]
+impl TokenOwnerProposing for SeasolverTokenOwnerFinder {
+    async fn find_candidate_owners(&self, _: H160) -> Result<Vec<H160>> {
+        Ok(self.owners.clone())
+    }
```

Comment:
There is a question here whether the addresses should be token-owner pairs. If you have a single address that owns most tokens then you don't care about the token-address but if you have a long list of addresses that each own only a single token then it would be wasteful to return all of them every time because that causes more work in the code that calls this code here because it has to actually find the balance.

The most correct way to implement might be to have a `HashMap<TokenAddress, OwnerAddress>` in this struct and configure it that way in the arguments too.

On the other hand if it's just one address at the moment then maybe implementing it that way isn't worth it.

Reply:
This is a good suggestion and scales better. Also, considering that this list is coming from memory, its likely to return very quickly and always get considered for every token (despite being the holder for just a few tokens), so I think it would be better to have a `HashMap<TokenAddress, OwnerAddress>` or even `HashMap<TokenAddress, Vec<OwnerAddress>>`.

Reply:
I am not sure about these suggestions, since both have pros and cons. In practice (Seasolver example) we have one owner and a list of rare tokens. Having `HashMap<TokenAddress, Vec<OwnerAddress>>` to support this requires more manual work to define arguments in configuration, but again, current solution is bad since we have necessary `eth_call` as @vkgnosis mentioned.
Ended up using `HashMap<TokenAddress, Vec<OwnerAddress>>`